nixos/modules/services/networking/wireguard.nix

```diff
@@ -153,35 +182,43 @@ let
   wgCommand = "${pkgs.wireguard}/bin/wg";
 
   generateUnit = name: values:
+    assert (values.privateKey != null || values.privateKeyFile != null);
     nameValuePair "wireguard-${name}"
       {
         description = "WireGuard Tunnel - ${name}";
         after = [ "network.target" ];
         wantedBy = [ "multi-user.target" ];
+        script = builtins.concatStringsSep "\n" (lib.flatten([
+          values.preSetup
+
+          ''
+          ${ipCommand} link del dev "${name}" || true
+          ${ipCommand} link add dev ${name} type wireguard
+          ${wgCommand} setconf ${name} /dev/stdin <<EOF
```

Comment:
I switched from script to ExecStart because ExecStart did not like my heredoc. It is like this because with a heredoc I could avoid having to create a temporary file and having to make sure it has locked down permissions.